enhancements/1186-spiffe-integration/README.md

```diff
@@ -0,0 +1,54 @@
+- **Non-Goals:** This proposal does not aim to present how Spiffe works, but rather focuses on the integration of Spiffe in the NFD project.
+
+## Proposal
+<img src="./nfd-spiffe.png" style="display: block; margin-left: auto; margin-right: auto;">
+As shown in the figure above, the proposed solution involves:
+
+1. Installation of Spiffe which is composed of Spire-agents as daemonset, and Spire-server as statefulset. Spiffe installation should be available in Helm as well as Kustomize.
+
+2. The cluster administration (who intends to use NFD), should initialize the Spire-server with the specific Spiffe Verifiable Identity Document ID a.k.a SVID. In fact, we need to have only one SVID shared between all of the nfd-worker workloads. Initialization commands should be provided in docs.
+
+3. After features discovery, nfd-workers need to get their private key from Spiffe and sign the NodeFeature CR spec with their private keys. The signature should be put in the annotations section of the NodeFeature CR. e.g.: `annotations.signature`
+
+4. When detecting new changes in the NodeFeature CR, the nfd-master should verify the signature in the CR by using the public key of the nfd-workers.
+
+Such a process would result in exposing only features created by the nfd-worker. In fact, when a workload/user changes manually the NodeFeature CR, updates won't be propagated to nodes.
+
+### Notes
+- The signature of NodeFeature CR spec is encoded in base64 to make it utf-8 encoded, since the signature itself is not utf-8 encoded, and might have special characters that can result in some errors.
+- The nfd-master does not need any sort of SVID as it does not need to sign data.
+- If users need to integrate 3rd party tools along with NFD, they need to create their respective SVIDs in the Spire-server, so that exposed data can be validated.
+- This feature should be noted as EXPERIMENTAL, and enabling/disabling Spiffe verification should be configurable for users via cmdline or configuration file.
+- Some tweaks that need to be done to the cluster in order to enable Spiffe that should be mentioned in docs. Example of Minikube configuration tweak: https://spiffe.io/docs/latest/try/getting-started-k8s/#considerations-when-using-minikube
+
+
+### Risks and Mitigations
+- Workloads communication with Spire-agent and Spire-server is done through a Unix socket, which should be mounted from the host to workloads, which is not really recommended from a security perspective.
+
+## Design Details
+- Create a Spiffe utility structure in `pkg/utils` that would be responsible for exposing the following functions:
+    - `NewSpiffeClient`: returns a new `workloadapi.Client` which allows us to communicate with the spire-server.
+    - `SignData`: returns a signature of a given spec.
```

Comment:
There probably are good utility/library functions for getting the signature(?) Do we need to put any details about that (e.g. hashing/signing algorithms?).